gradle/scala_211_212.gradle

```diff
@@ -9,8 +9,8 @@ crossBuild {
 }
 
 dependencies {
```

Comment:
So if I understand this change correctly, it changes the default to 2.11? If yes, why was this change made? Maybe I am still not understanding how this propagates down to project dependencies.

Reply:
As I already mentioned, when you run `./gradlew clean build ..` It only builds default scala version and test. If the default version is not aligned the tests will fail. `build` is not triggering any crossbuild tasks. This can be achieved using `publish...` tasks and the specified cross build tasks as can be seen when running `./gradlew tasks`. the tasks for cross build project are dynamic and based on the actual project - how many cross build variants declared, if publish is used etc

Reply:
OK, now you lost me completely when you said that you need to run a `publish` task for project dependencies. This is how Maven treats module dependencies. In Maven, you have to publish a JAR of a module to the Maven local repository before you can use it as a module dependency. And that's why you the `install` goal. Gradle resolves project dependencies dynamically by referencing the JAR file of project it depends on and builds it automatically. Publishing is not involved at all.

Maybe that's why we are misunderstanding each other?

Reply:
The tasks dependency graph in Gradle is built and as implicitly all cross built publish tasks are dependent on cross built artifacts, Gradle infers correctly that publish task is dependent on cross build jar task. Hence when you call publish you call down the road the relevant cross built tasks.

Reply:
I think we are talking about two different use cases here:

1. Build my project with all variants and project dependencies (including its combinations). For that you'd call `./gradle build` for example or `./gradlew assemble`. I do not want to publish anything so the `publish` task isn't even involved.
2. Same as 1. but publish artifacts to a binary repository where necessary. I'd call `./gradlew publish` for that.

I am primarily asking about 1. So I am still not understanding how the project dependencies would be declared or resolved correctly for all variants. From what I understand from your previous comment, project dependencies can only be resolved for the default Scala version. That won't solve my use case. It's OK if that's a limitation of the plugin. I just want to make sure I am understanding correctly.

Reply:
> Build my project with all variants and project dependencies (including its combinations). For that you'd call ./gradle build for example or ./gradlew assemble. I do not want to publish anything so the publish task isn't even involved.

Not correct, see the plugin documentation for the different tasks the plugin adds.

> Same as 1. but publish artifacts to a binary repository where necessary. I'd call ./gradlew publish for that.

see my explanation on Gradle tasks dependency graph

> I am primarily asking about 1. So I am still not understanding how the project dependencies would be declared or resolved correctly for all variants. From what I understand from your previous comment, project dependencies can only be resolved for the default

If I can suggest skip for now the `default` technicalities and concentrate on the tasks the plugin adds. They them out after enlisting them with gradle tasks